fili-core/src/test/groovy/com/yahoo/bard/webservice/data/config/metric/makers/MetricMakerSpec.groovy

```diff
@@ -81,32 +158,22 @@ class MetricMakerSpec extends Specification {
         }
     }
 
-    def setup(){
-        maker = getInstance()
-        //Initializing the Sketch field converter
-        FieldConverterSupplier.sketchConverter = new SketchFieldConverter();
-    }
-
-    def cleanupSpec() {
-        FieldConverterSupplier.sketchConverter = converter
-    }
-
     def "When a metric has too few dependent metrics, an exception is thrown."(){
         given: "One fewer dependent metrics than this metric depends on"
-        List<String> dependentMetricNames = buildDependentMetricNames(maker.getDependentMetricsRequired()-1)
-        maker.metrics.putAll(makeEmptyMetrics(dependentMetricNames))
+        List<String> dependentMetricNames = buildDependentMetricNames(DEPENDENT_METRICS - 1)
+        dictionary.putAll(makeEmptyMetrics(dependentMetricNames))
 
         when:
-        maker.make(METRIC_NAME, dependentMetricNames)
+        getMakerInstance().make(METRIC_NAME, dependentMetricNames)
 
         then:
         thrown(IllegalArgumentException)
```

Comment:
I personally prefer adding the error message to the exception check, just to make sure that it is the correct exception thrown.